File: app/src/main/java/org/michaelbel/moviemade/rest/api/service/KEYWORDS.java
package org.michaelbel.moviemade.rest.api.service;

import org.michaelbel.moviemade.rest.model.v3.Keyword;
import org.michaelbel.moviemade.rest.response.MoviesResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface KEYWORDS {

    @GET("keyword/{keyword_id}?")
    Observable<Keyword> getDetails(
        @Path("keyword_id") int id,
        @Query("api_key") String apiKey
    );

    @GET("keyword/{keyword_id}/movies")
    Observable<MoviesResponse> getMovies(
        @Path("keyword_id") int id,
        @Query("api_key") String apiKey,
        @Query("language") String language,
        @Query("include_adult") boolean adult,
        @Query("page") int page
    );
}

File: app/src/main/java/org/michaelbel/moviemade/data/service/KEYWORDS.kt
package org.michaelbel.moviemade.data.service

import io.reactivex.Observable
import org.michaelbel.moviemade.rest.model.v3.Keyword
import org.michaelbel.moviemade.rest.response.MoviesResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface KEYWORDS {

    @GET("keyword/{keyword_id}?")
    fun getDetails(
        @Path("keyword_id") id: Int,
        @Query("api_key") apiKey: String
    ): Observable<Keyword>

    @GET("keyword/{keyword_id}/movies")
    fun getMovies(
        @Path("keyword_id") id: Int,
        @Query("api_key") apiKey: String,
        @Query("language") language: String,
        @Query("include_adult") adult: Boolean,
        @Query("page") page: Int
    ): Observable<MoviesResponse>
}